### ISSUE TITLE
Recompiling each time after I restart the kernel in Linux.

### ISSUE THREAD

**Issue by baiguoname** [evcxr/evcxr#320]
I am using evcxr_jupyter(in vscode jupyter extention) in Linux(WSL in windows), and I set the `export EVCXR_TMPDIR=/root/evcxr_tmpdir`. But it seems the setting didn't work, the compiling code is still in the system tmp directory.
Each time I restart the kernel, evcxr will start a new compilling.
On the other hand, I apply the similar process above on windows, everything work fine.
Am I doing something wrong?

**Comment by davidlattimore** [comment]
I've heard of WSL, but haven't used (since I don't have Windows), so my knowledge on this is very limited. Is it normal to do everything as root on WSL?

Some thoughts of things to check:

- That EVCXR_TMPDIR is definitely set in the running kernel. e.g. run `std::env::var("EVCXR_TMPDIR")`
- Look to see what's in the directory while the kernel is running. There should be a `Cargo.toml`, `target` directory etc.
- You could try writing a file to the directory. e.g.: `std::fs::write(std::path::Path::new(&std::env::var("EVCXR_TMPDIR")?).join("test.txt"), "test")?;` then check to see if it's still there after you restart the kernel.
- If the directory gets deleted, when it is getting deleted - e.g. is it when the old kernel shuts down, when the new kernel starts up, or some other time?
- How are you determining that everything is being recompiled? Is this based on the compilation messages being logged to a shell where `jupyter-notebook` was launched?

**Comment by bjorn3** [comment]
And do you use WSL1 or WSL2?

**Comment by baiguoname** [comment]
> That EVCXR_TMPDIR is definitely set in the running kernel. e.g. run std::env::var("EVCXR_TMPDIR")
Thanks for your guidance, by following that, it turns out that I didn't set the environmental variable corectlly.
I did this in jupyter, and the output show `Err(NotPresent)`, but in Linux terminal the result of `echo $EVCXR_TMPDIR` show the right result `"/root/evcxr_dir"`. I thought it may be a problem about the system, so I re-install the WSL. Then the output show `ok(/root/evcxr_dir)` and everything works fine.
Then I tested the performance difference between using evcxt_jupyter in Windows and in Linux, the result is almost as expected by [this discussion](https://github.com/evcxr/evcxr/discussions/313). With using the same code, the startup time in Linux took 18s, and that in Windows took 30s. After startup, the simple code, like `let a = 1;`, had a elapsed time about 0.5 in Linux, and had a elapsed time about 1.2s in Windows.

**Comment by baiguoname** [comment]
> * That EVCXR_TMPDIR is definitely set in the running kernel. e.g. run `std::env::var("EVCXR_TMPDIR")`

Thanks for your guidance, by following that, it turns out that I didn't set the environmental variable corectlly.
I did this in jupyter, and the output show Err(NotPresent), but in Linux terminal the result of echo $EVCXR_TMPDIR show the right result "/root/evcxr_dir". I thought it may be a problem about the system, so I re-install the WSL. Then the output show ok(/root/evcxr_dir) and everything works fine.
Then I tested the performance difference between using evcxt_jupyter in Windows and in Linux, the result is almost as expected by https://github.com/evcxr/evcxr/discussions/313. With using the same code, the startup time in Linux took 18s, and that in Windows took 30s. After startup, the simple code, like let a = 1;, had a elapsed time about 0.5 in Linux, and had a elapsed time about 1.2s in Windows.
